### ISSUE TITLE
Does not work with Fitbit One

### ISSUE THREAD

**Issue by snrmwg** [paulburton/fitbitd#2]
Can't get it to work with my Fitbit One. Maybe because it has different USB vendor:product ID?
You seem to have hardcoded 10c4:84c4. When I do a lsusb then I get 2687:fb01 for the Fitbit One.

**Comment by paulburton** [comment]
That's expected. The new one & zip trackers are significantly different to the older ones, and I only have the older fitbit ultra so I can't implement support for the newer trackers right now.

**Comment by snrmwg** [comment]
is there any documentation or specifications? maybe I could add support for newer tracker.

**Comment by paulburton** [comment]
Nope, if only it were that easy. You'd need to reverse engineer from the official fitbit client.

**Comment by snrmwg** [comment]
holy moly :) can you suggest tools to reverse engineer the protocol? I develop in linux and have windows xp instance in a virtual box.

**Comment by paulburton** [comment]
Wireshark is useful, I used it on my (linux-running) machine with windows in virtualbox & captured both the usb data & network traffic from the fitbit client without too much hassle. As the new one & zip trackers are bluetooth based though I'm guessing the fitbit dongle will just be a relatively standard bluetooth dongle of some sort & it would make sense to capture traffic at the bluetooth rather than usb level (so other bluetooth 4 devices could sync without the fitbit dongle). Of course I don't own one so could be completely wrong.

**Comment by anthroprose** [comment]
The fitbit one does not register with lsusb or any other standard USB type diagnostics.

Assuming this means there is no USB interface other than 5V and everything happens over bluetooth.

If there is any interest in expanding this daemon to include Bluetooth I can/will gladly capture/debug any bluetooth traffic with my unit however I have little/no bluetooth dev experience.

**Comment by drewp** [comment]
Here's a tiny bit more data about the Zip dongle on linux:
[36459.959919] usb 1-1.5: New USB device found, idVendor=2687, idProduct=fb01
[36459.959929] usb 1-1.5: New USB device strings: Mfr=1, Product=2, SerialNumber=3
[36459.959935] usb 1-1.5: Product: Fitbit Base Station
[36459.959940] usb 1-1.5: Manufacturer: Fitbit Inc.
[36459.959945] usb 1-1.5: SerialNumber: \xffffffef\xffffffbf...
[36459.963839] hid-generic 0003:2687:FB01.000C: hiddev0,hidraw1: USB HID v1.11 Device [Fitbit Inc. Fitbit Base Station] on usb-0000:00:1a.0-1.5/input0
[36459.967054] hid-generic 0003:2687:FB01.000D: hiddev0,hidraw2: USB HID v1.11 Device [Fitbit Inc. Fitbit Base Station] on usb-0000:00:1a.0-1.5/input1

**Comment by n8willis** [comment]
And, to accompany that, here's the One dongle:
[ 2625.616219] usb 1-5.4.4: new full-speed USB device number 13 using ehci_hcd
[ 2625.734242] generic-usb 0003:2687:FB01.0003: hiddev0,hidraw2: USB HID v1.11 Device [Fitbit Inc. Fitbit Base Station] on usb-0000:00:02.1-5.4.4/input0
[ 2625.737462] generic-usb 0003:2687:FB01.0004: hiddev0,hidraw3: USB HID v1.11 Device [Fitbit Inc. Fitbit Base Station] on usb-0000:00:02.1-5.4.4/input1

That said, evidently the dongle is a Bluetooth 4.0 Low-Energy device; dunno how adversely that might affect things.

**Comment by Rudd-O** [comment]
keep me posted. for now I cannot use my fitbit because of thir.

**Comment by JohnConnett** [comment]
I have a Fitbit One™. Here are some details obtained using Device Manager on Windows Vista Business 32-bit, SP2. From the driver tab of properties:

USB Composite Device
Driver Provider: Microsoft
Driver Date: 21/06/2006
Driver Version: 6.0.6002.18005
Driver Signer: microsoft windows

From the details tab of properties:

Hardware Ids:
USB\VID_2687&PID_FB01&REV_0100
USB\VID_2687&PID_FB01
Children:
USB\VID_2687&PID_FB01&MI_00\6&12f0c97d&0&0000
USB\VID_2687&PID_FB01&MI_01\6&12f0c97d&0&0001

So the Fitbit One™ Wireless USB Dongle for Zip™ & One™ Trackers appears to use the standard Microsoft USB Composite Device driver and has a couple of child devices.

Does anyone know if the Fitbit One™ Wireless USB Dongle can be used as a general purpose Bluetooth™ 4.0 device?

I assume that as the Fitbit One™ can sync to modern Apple iStuff it must be visible as a Bluetooth™ 4.0 device? Unfortunately, I don't have a smart phone or anything else that supports Bluetooth™ 4.0 to try it with ...

Would love to be able to use it with Linux and to be able to set the silent wake alarm (currently Apple iStuff only?).

I would be happy to try testing any changes to support the Fitbit One™.

**Comment by gyhor** [comment]
maybe the information from this blog could help?
http://hacksbypete.blogspot.ca/

**Comment by crazedsanity** [comment]
I have a Fitbit One as well, and I'm definitely willing to help wherever possible.  My primary OS is Ubuntu Linux (13.04, x86_64), but I also have Win7 Ultimate in a VirtualBox instance, and Windows 7 Professional on a different partition (it's a dual-boot).

For the blog reference above, from gyhor, try this: [ http://hacksbypete.blogspot.com/search?q=fitbit ].  It does look like there's quite a bit of useful information in there.

If anybody has tips on where I should start for helping, let me know.  I'm a consultant by trade, and a hacker by hobby.

**Comment by vorburger** [comment]
Note this issue https://github.com/openyou/libfitbit/issues/46 (on libfitbit), pointing to https://bitbucket.org/benallard/galileo.
